feature_engine/creation/decision_tree_features.py

```diff
@@ -0,0 +1,422 @@
+    _drop_original_docstring,
+)
+
+from feature_engine._docstrings.substitute import Substitution
+from feature_engine.dataframe_checks import (
+    _check_contains_inf,
+    _check_contains_na,
+    _check_X_matches_training_df,
+    check_X_y,
+)
+from feature_engine.variable_manipulation import (
+    _check_input_parameter_variables,
+    _find_or_check_numerical_variables,
+)
+
+
+@Substitution(
+    variables=_variables_numerical_docstring,
+    drop_original=_drop_original_docstring,
+    variables_=_variables_attribute_docstring,
+    feature_names_in_=_feature_names_in_docstring,
+    n_features_in_=_n_features_in_docstring,
+    fit_transform=_fit_transform_docstring,
+)
+class DecisionTreeFeatures(BaseEstimator, TransformerMixin):
+    """
+    DecisionTreeFeatures() creates new variables by combining features using decision
+    trees. DecisionTreeFeatures() will add as variables, the predictions of decision
+    trees trained on 1, 2, 3 or more variables.
+
+    DecisionTreeFeatures() is a useful method, when using linear models, because it
+    captures non-linear relationships between features and the target by using trees.
+
+    DecisionTreeFeatures() works only with numerical variables. Categorical variables
+    need to be encoded into numbers first.
+
+    Missing data should be imputed before using this transformer.
+
+    More details in the :ref:`User Guide <dtrees_features>`.
+
+    Parameters
+    ----------
+    {variables}
+
+    strategy: integer, list or tuple, default=None
+        Indicates how to combine the features in `variables` to train the decision
+        trees. If None, then decision trees will be created with all possible feature
+        combinations, from 1 to the total number of features. If integer, then all
+        possible feature combinations from 1 to strategy will be used to train the
+        trees. If list of integers, then all combinations of features of group size
+        indicated in the list will be created. If tuple with tuples of variable names,
+        then the decision trees will be created using the variables specified in the
+        tuples.
+
+        **Examples**
+
+
+        The parameter `variables` takes ["var_A", "var_B", "var_C"]:
+
+        - `strategy=1` returns 3 new features based on single variable decision trees.
+        - `strategy=2` returns 6 new features based on single and double variable decision
+        trees
+        - `strategy=3` returns 7 new features based on single, double and triple variable
+        decision trees.
+        - `strategy=None` returns 7 new features based on single, double and triple
+        variable decision trees.
+        - `strategy=[1,2] returns 6 new features based on single and double variable
+        decision trees
+        - `stragety=(("var_A", "var_D"), ("var_B", "var_C")), returns 2 new features
+        based on the combinations in the tuples.
+
+        cv: int, cross-validation generator or an iterable, default=3
```

Comment:
brought forward the docstrings from DecisionTreeDiscretiser for the parameters for grid search and cross val